pkg/ipamd/datastore/data_store.go

```diff
@@ -145,38 +148,57 @@ type ENI struct {
 	IsTrunk bool
 	// IsEFA indicates whether this ENI is tagged as an EFA
 	IsEFA bool
-	// IsPDEnabled indicates whether prefix delegation is enabled
-	IsPDEnabled bool
 	// DeviceNumber is the device number of ENI (0 means the primary ENI)
 	DeviceNumber int
 	// IPv4Addresses shows whether each address is assigned, the key is IP address, which must
 	// be in dot-decimal notation with no leading zeros and no whitespace(eg: "10.1.0.253")
-	IPv4Addresses map[string]*AddressInfo
-	// Key is the prefix and value is prefix, used IPs under the prefix
-	IPv4Prefixes map[string]*ENIPrefix
+	// Key is the IP address
+	AvailableIPv4Cidrs map[string]*AssignedIPv4Addresses
 }
 
 // AddressInfo contains information about an IP, Exported fields will be marshaled for introspection.
 type AddressInfo struct {
 	IPAMKey        IPAMKey
 	Address        string
 	UnassignedTime time.Time
-	Prefix         net.IP
+	Prefix         net.IPNet
```

Comment:
should this be called Cidr for consistency?

Reply:
Actually, I will delete this, we are not using it anymore.